packages/salesforcedx-vscode-core/src/commands/util/parameterGatherers.ts

```diff
@@ -120,7 +121,15 @@ export class SelectFileName
     CancelResponse | ContinueResponse<{ fileName: string }>
   > {
     const fileNameInputOptions = {
-      prompt: nls.localize('parameter_gatherer_enter_file_name')
+      prompt: nls.localize('parameter_gatherer_enter_file_name'),
+      validateInput: value => {
+        try {
+          CreateUtil.checkInputs(value);
```

Comment:
Need to make sure this check function isn't too restrictive. What it's checking for (in other words, things that cause it to throw an error):

- Non AlphaNumeric characters
- Not starting with a letter
- Ending with an underscore
- Containing a double underscore

Reply:
Non AlphaNumeric characters means only letter, numbers, and underscores are okay. Hyphens, spaces, slashes, etc will throw an error.